diff --git a/nvchecker_source/cmd.py b/nvchecker_source/cmd.py
index bcd0151..dbfc7cd 100644
--- a/nvchecker_source/cmd.py
+++ b/nvchecker_source/cmd.py
@@ -56,4 +56,5 @@ async def get_version(
   name, conf, *, cache, keymanager=None
 ):
   cmd = conf['cmd']
-  return await cache.get(cmd, partial(run_cmd, name))
+  timeout = conf.get('timeout', 300)
+  return await cache.get(cmd, partial(run_cmd, name, timeout=timeout))
