diff --git a/app/server/fireshare/__init__.py b/app/server/fireshare/__init__.py
index 93768f5..4c679d6 100644
--- a/app/server/fireshare/__init__.py
+++ b/app/server/fireshare/__init__.py
@@ -81,7 +81,7 @@ def create_app(init_schedule=False):
     app.config['LDAP_ADMIN_GROUP'] = os.getenv("LDAP_ADMIN_GROUP")
 
     app.config['SQLALCHEMY_DATABASE_URI'] = f'sqlite:///{app.config["DATA_DIRECTORY"]}/db.sqlite'
-    app.config['SCHEDULED_JOBS_DATABASE_URI'] = f'sqlite:///jobs.sqlite'
+    app.config['SCHEDULED_JOBS_DATABASE_URI'] = f'sqlite:///{app.config["DATA_DIRECTORY"]}/jobs.sqlite'
     app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
     app.config['INIT_SCHEDULE'] = init_schedule
     app.config['MINUTES_BETWEEN_VIDEO_SCANS'] = int(os.getenv('MINUTES_BETWEEN_VIDEO_SCANS', '5'))
diff --git a/app/server/fireshare/main.py b/app/server/fireshare/main.py
index b86727c..b36fa19 100644
--- a/app/server/fireshare/main.py
+++ b/app/server/fireshare/main.py
@@ -12,7 +12,7 @@ main = Blueprint('main', __name__, template_folder=templates_path)
 
 CORS(main, supports_credentials=True)
 
-@main.before_app_first_request
+@main.before_app_request
 def before_first_request():
     # Create the admin user if it doesn't already exist
     admin = User.query.filter_by(admin=True, ldap=False).first()
@@ -24,7 +24,7 @@ def before_first_request():
         db.session.commit()
     if admin and not check_password_hash(admin.password, current_app.config['ADMIN_PASSWORD']):
         row = db.session.query(User).filter_by(admin=True, ldap=False).first()
-        row.password = generate_password_hash(current_app.config['ADMIN_PASSWORD'], method='sha256')
+        row.password = generate_password_hash(current_app.config['ADMIN_PASSWORD'], method='pbkdf2:sha256')
         db.session.commit()
     if admin and current_app.config['ADMIN_USERNAME'] and admin.username != current_app.config['ADMIN_USERNAME']:
         row = db.session.query(User).filter_by(admin=True, ldap=False).first()
diff --git a/app/server/requirements.txt b/app/server/requirements.txt
index eb334a0..f46eb86 100644
--- a/app/server/requirements.txt
+++ b/app/server/requirements.txt
@@ -1,24 +1,24 @@
-click==8.1.3
-ffmpeg-python==0.2.0
-Flask==2.1.2
-Flask-Cors==3.0.10
-Flask-Login==0.6.1
-Flask-Migrate==3.1.0
-Flask-SQLAlchemy==2.5.1
-Flask-WTF==1.0.1
-future==0.18.2
-greenlet==3.0.3
-gunicorn==20.1.0
-importlib-metadata==4.11.3
-itsdangerous==2.1.2
-Jinja2==3.1.2
-MarkupSafe==2.1.1
-six==1.16.0
-SQLAlchemy==1.4.36
-Werkzeug==2.1.2
-WTForms==3.0.1
-zipp==3.8.0
-xxhash==3.0.0
-apscheduler==3.9.1
-python-ldap==3.4.3
-requests==2.27.1
\ No newline at end of file
+click
+ffmpeg-python
+Flask
+Flask-Cors
+Flask-Login
+Flask-Migrate
+Flask-SQLAlchemy
+Flask-WTF
+future
+greenlet
+gunicorn
+importlib-metadata
+itsdangerous
+Jinja2
+MarkupSafe
+six
+SQLAlchemy
+Werkzeug
+WTForms
+zipp
+xxhash
+apscheduler
+python-ldap
+requests
\ No newline at end of file
